### ISSUE TITLE
Current SecurityContext settings doesnot allow to run it secure on EKS

### ISSUE THREAD

**Issue by dschunack** [jfrog/charts#1838]
**BUG REPORT** (choose one):

**Version of Helm and Kubernetes**: latest version [7.71.3 release](https://github.com/jfrog/charts/commit/2d52843fb754c309d6de7e028f2f7096ca484317) on EKS 1.28 with Kyverno. The following Policies are enforced.

[Kyverno Policies](https://kyverno.io/policies/?policytypes=Pod%2520Security%2520Standards%2520%28Baseline%29%2BPod%2520Security%2520Standards%2520%28Restricted%29%2BSecurity)
[Kyverno NonRootGroup](https://kyverno.io/policies/other/rec-req/require-non-root-groups/require-non-root-groups/)

**Which chart**: latest version [7.71.3 release](https://github.com/jfrog/charts/commit/2d52843fb754c309d6de7e028f2f7096ca484317)

**Which product license** (Enterprise/Pro/oss):
Enterprise

**JFrog support reference** (if already raised with support team): 
we will add it later

**What happened**:
It's not possible at the moment run artifactory on EKS with Kyverno Policies that doesn't allow it to run PODs with UID and GID = 0.
The problem is the following change, that was implemented to fix issues on OpenShift, but it blocks us to install it on EKS.
It's not possible to set the GID and fsGroup, due to this [change](https://github.com/jfrog/charts/blob/7a7916b55001ae2f12cdad699de717536720b596/stable/artifactory/templates/artifactory-statefulset.yaml#L94-L98). It make no sense to enable this only for OpenShift.
The Container is able to run without a UID and GID = 0, but why is this not the standard?

As you know, we all want to keep our environment secure and you have the possibility to run Artifactroy by default with a User UID/GID and not with the Root UID/GID.

Our proposal is to run Artifactory by default with a User UID/GID and to revert the following [Changes](https://github.com/search?q=repo%3Ajfrog%2Fcharts%20artifactory.isOpenshiftCompatible&type=code) and to set the SecurityContext by default like this PR  #1744 .

This helps all they want to keep there k8s secure and to Fulfill the [PSS Policies](https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted) for K8S Solutions and not only on OpenShift.

**What you expected to happen**:
Implement a way to run Artifactory Secure on all K8S Solutions like EKS/AKS/GKE and OpenShift.

**How to reproduce it** (as minimally and precisely as possible):
Use EKS with Kyverno or any solution that fulfills the PSS  and try to install Artifactory.

At the moment it will fail, it's not possible to set the correct GID, it's blocked by the following [Code](https://github.com/search?q=repo%3Ajfrog%2Fcharts%20artifactory.isOpenshiftCompatible&type=code)

**Comment by chukka** [comment]
Thanks for raising this , we are currently fixing this internally and will be able to release a patch in two weeks

**Comment by Logeshwarsn** [comment]
Update : We have fix ready and it will be include in the next patch 7.71.x

**Comment by gjlam95** [comment]
> Update : We have fix ready and it will be include in the next patch 7.71.x

Hi developers, do note that the same issue is found in the latest xray chart as well. Lines of code affected [here](https://github.com/jfrog/charts/blob/b087778381fc6515f6fc579baed363b5ade271f7/stable/xray/templates/xray-statefulset.yaml#L88C1-L95)

**Comment by chukka** [comment]
Update : This fix release  has been delayed by atleast couple of weeks

**Comment by rahulsadanandan** [comment]
@gjlam95 The fix for the Xray will be implemented in version 3.90.x.

**Comment by rahulsadanandan** [comment]
Sorry for the delay. The Xray fix has been merged into version 3.91.x.
